### PR TITLE
Convert time column in PlainFieldExtractor

### PR DESCRIPTION
**Pull request by fx19880617**

### CODE REVIEW

**Comment by kishoreg** [comment]
is this diff from master? looks like diff from the previous PR. Can you confirm. This is looking better.

**Comment by fx19880617** [comment]
Yes, this is one commit ahead master.

File: pinot-common/src/main/java/com/linkedin/pinot/common/utils/time/GeneralTimeConverter.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.common.utils.time;
+
+import com.linkedin.pinot.common.data.TimeGranularitySpec;
+
+public class GeneralTimeConverter implements TimeConverter {
```
Comment (kishoreg) [inline pinot-common/src/main/java/com/linkedin/pinot/common/utils/time/GeneralTimeConverter.java]: rename to GenericTimeConvertor or DefaultTimeConverter

File: pinot-core/src/main/java/com/linkedin/pinot/core/data/extractors/PlainFieldExtractor.java [hunk 2]
```diff
@@ -113,6 +129,15 @@ public GenericRow transform(GenericRow row) {
...
       boolean hasConversion = false;
       for (String column : _schema.getColumnNames()) {
+        // Adding outgoing time column to fieldMap;
+        if (column.equals(_schema.getTimeColumnName())) {
+          Object incomingTimeValue = row.getValue(_incomingTimeColumnName);
+          if (incomingTimeValue == null) {
+            incomingTimeValue = _schema.getTimeFieldSpec().getDefaultNullValue();
+          }
+          fieldMap.put(_outGoingTimeColumnName, _timeConverter.convert(incomingTimeValue));
+          continue;
```
Comment (kishoreg) [inline pinot-core/src/main/java/com/linkedin/pinot/core/data/extractors/PlainFieldExtractor.java]: remove this, the rest of the code should handle this correct, it has lot of things like setting default value, counting error rows etc

File: pinot-core/src/main/java/com/linkedin/pinot/core/data/extractors/PlainFieldExtractor.java [hunk 3]
```diff
@@ -113,6 +129,15 @@ public GenericRow transform(GenericRow row) {
       boolean hasNull = false;
       boolean hasConversion = false;
       for (String column : _schema.getColumnNames()) {
+        // Adding outgoing time column to fieldMap;
+        if (column.equals(_schema.getTimeColumnName())) {
+          Object incomingTimeValue = row.getValue(_incomingTimeColumnName);
+          if (incomingTimeValue == null) {
```
Comment (kishoreg) [inline pinot-core/src/main/java/com/linkedin/pinot/core/data/extractors/PlainFieldExtractor.java]: you can remove this from here, next section should take care of this

File: pinot-common/src/main/java/com/linkedin/pinot/common/utils/time/GeneralTimeConverter.java (hunk 1 above)
Comment (kishoreg) [inline pinot-common/src/main/java/com/linkedin/pinot/common/utils/time/GeneralTimeConverter.java]: Add a standalone unit test for this class.

**Comment by kishoreg** [comment]
LGTM. Minor comments.

File: pinot-core/src/main/java/com/linkedin/pinot/core/data/extractors/PlainFieldExtractor.java (hunk 2 above)
Comment (fx19880617) [inline pinot-core/src/main/java/com/linkedin/pinot/core/data/extractors/PlainFieldExtractor.java]: _schema.getTimeColumnName() will give outgoing time column name, which is not in GenericRow. 
So will check here to get value first, then do timeConverter just before setting fieldMap

**Comment by kishoreg** [comment]
Merged.

**Pull request closed without merging**